src/Magnum/Platform/WindowlessEglApplication.h

```diff
@@ -261,17 +262,44 @@ class WindowlessEglContext::Configuration {
          *
          * The device ID is expected to be smaller than the count of devices
          * reported by EGL. When using @ref WindowlessEglApplication, this is
-         * also exposed as a `--magnum-device` command-line option and a
-         * `MAGNUM_DEVICE` environment variable. If @ref setSharedContext() is
+         * also exposed as a `--magnum-device`/`--magnum-cuda-device` command-line option and a
+         * `MAGNUM_DEVICE`/`MAGNUM_CUDA_DEVICE` environment variable. If @ref setSharedContext() is
          * set, this value is ignored and the device is picked to be the same
          * as in the shared context instead.
+         *
+         * Use @ref deviceIsCuda to check if the device corresponds to a CUDA device
+         *
          * @requires_gles Device selection is not available in WebGL.
          */
         Configuration& setDevice(UnsignedInt id) {
             _device = id;
             return *this;
         }
 
+        /**
+         * @brief Whether or not the device is a CUDA device
+         *
+         * @note Requires an NVIDIA GPU/CUDA
+         * @requires_gles Device selection is not available in WebGL.
+         */
+        bool deviceIsCuda() const { return _deviceIsCuda; }
+
+
+        /**
+         * @brief Set whether or not the device is a CUDA device
+         * @return Reference to self (for method chaining)
+         *
+         * When using @ref WindowlessEglApplication, this is will be set
+         * to true if `--magnum-cuda-device` or `MAGNUM_CUDA_DEVICE` are
+         * specified
+         *
+         * @requires_gles Device selection is not available in WebGL.
+         */
+        Configuration& setDeviceIsCuda(const bool isCuda) {
+            _deviceIsCuda = isCuda;
+            return *this;
+        }
```

Comment:
I would instead add a pair of `cudaDevice()` and `setCudaDevice()`. That makes it closer to the pair of `--magnum-device` / `--magnum-cuda-device` options and makes it easier on when populating the config from code as well, since all you have to do is call just `setCudaDevice()` instead of calling both `setDevice()` and `setCudaDevice()`.